components/sup/src/manager/commands.rs

```diff
@@ -185,60 +184,39 @@ pub fn service_load(mgr: &ManagerState,
-
-            // We don't have any record of this thing; let's set it up!
-            //
-            // If a package exists on disk that satisfies the
-            // desired package identifier, it will be used;
-            // otherwise, we'll install the latest suitable
-            // version from the specified Builder channel.
-            util::pkg::satisfy_or_install(req, &source, &bldr_url, &bldr_channel)?;
-
-            mgr.cfg.save_spec_for(&spec)?;
-            req.info(format!("The {} service was successfully loaded", spec.ident))?;
+    let mut spec = if let Some(spec) = mgr.cfg.spec_for_ident(source.as_ref()) {
+        let force = opts.force.unwrap_or(false);
+        // We've seen this service  before. Thus `load`
+        // basically acts as a way to edit spec files on the
+        // command line. As a result, we a) check that you
+        // *really* meant to change an existing spec, and b) DO
+        // NOT download a potentially new version of the package
+        // in question
+        if !force {
+            return Err(net::err(ErrCode::Conflict,
+                                format!("Service already loaded, unload '{}' \
+                                         and try again",
+                                        ident)));
         }
-        Some(mut spec) => {
-            // We've seen this service  before. Thus `load`
-            // basically acts as a way to edit spec files on the
-            // command line. As a result, we a) check that you
-            // *really* meant to change an existing spec, and b) DO
-            // NOT download a potentially new version of the package
-            // in question
-
-            if !force {
-                return Err(net::err(ErrCode::Conflict,
-                                    format!("Service already loaded, unload '{}' \
-                                             and try again",
-                                            ident)));
-            }
-
-            opts.into_spec(&mut spec);
+        spec
+    } else {
+        ServiceSpec::default()
+    };
 
-            // Only install if we don't have something
-            // locally; otherwise you could potentially
-            // upgrade each time you load.
-            //
-            // Also make sure you're pulling from where you're
-            // supposed to be pulling from!
-            util::pkg::satisfy_or_install(req, &source, &spec.bldr_url, &spec.channel)?;
+    opts.into_spec(&mut spec);
 
-            mgr.cfg.save_spec_for(&spec)?;
-            req.info(format!("The {} service was successfully loaded", spec.ident))?;
-        }
+    // If a package exists on disk that satisfies the desired package identifier, it will be used.
+    // Otherwise, we'll install the latest suitable version from the specified Builder channel.
+    let package = util::pkg::satisfy_or_install(req, &source, &spec.bldr_url, &spec.channel)?;
+    if let Err(e) = spec.validate(&package) {
```

Comment:
Thanks for catching this. I agree that would have been very confusing. I was doing this to make the error code be `InvalidPayload` instead of `Internal`, but the proper way to do this is just fix the `From` implementation. This [commit](https://github.com/habitat-sh/habitat/pull/6730/commits/48db0b9e4ccd657f374ac08f8d3c8308bbdfb4aa) should do it.